From 89f5fa47476eda56402e29fff3c5097f5c2a1e19 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Mon, 3 Dec 2018 16:47:21 -0500
Subject: [PATCH] dm: call blk_queue_split() to impose device limits on bios
Git-commit: 89f5fa47476eda56402e29fff3c5097f5c2a1e19
Patch-mainline: v4.20-rc7
References: git-fixes

Otherwise the incoming bios, of various types, won't be shaped based on
the DM device's advertised limits.

(Coly Li: rebase for SLE15 kernel)

Depends-on: af67c31fba ("blk: remove bio_set arg from blk_queue_split()")
Fixes: 744889b7cb ("block: don't deal with discard limit in blkdev_issue_discard()")
Cc: stable@vger.kernel.org
Signed-off-by: Mike Snitzer <snitzer@redhat.com>
Signed-off-by: Coly Li <colyli@suse.de>
---
 drivers/md/dm.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/md/dm.c
+++ b/drivers/md/dm.c
@@ -1471,6 +1471,8 @@ static void __split_and_process_bio(stru
 		return;
 	}
 
+	blk_queue_split(md->queue, &bio);
+
 	ci.map = map;
 	ci.md = md;
 	ci.io = alloc_io(md);
